WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Success card/MigrationSuccessCardView.swift

```diff
@@ -73,13 +73,20 @@ class MigrationSuccessCardView: UIView {
     }
 }
 
-// TODO: This extension is temporary, and should be replaced by the actual condition to check, and placed in the proper location
+// Ideally, this logic should be handled elsewhere. But since the whole migration feature is temporary
+// Perhaps that's not worth the trouble.
+//
+// TODO: Remove `shouldShowMigrationSuccessCard` when the migration feature is no longer needed
 extension MigrationSuccessCardView {
-    @objc
-    static var shouldShowMigrationSuccessCard: Bool {
 
-        AppConfiguration.isJetpack && showCard
+    @objc static var shouldShowMigrationSuccessCard: Bool {
```

Comment:
Yes that's correct. Good catch!

It would be great to remove those tracks from `getWordPressInstallationState()` and move them elsewhere. Because that function is supposed to be responsible only for returning the WordPress installation state. Tracking events in that function is an unexpected side-effect. 

It seems the `getWordPressInstallationState()` was never used before, so I guess it's fine to remove those tracks, what do you say?

<img width="600" src="https://user-images.githubusercontent.com/9609223/205136515-febbaf8b-9072-46ae-a0ea-fcb4cc9d363d.png">

Reply:
I don't think we should remove the tracks, rather, as you say, move them elsewhere: as long as they are put in a place where we correctly track the compatibility status once, I'm fine with that, and I think it should be addressed in this same PR.

Reply:
👋 Hey @salimbraksa and @Gio2018

> Because that function is supposed to be responsible only for returning the WordPress installation state.

I agree with this - I think most callers will assume there won't be any side effects generated from this call.

We can be sure to move the calls where we make the check outlined here: https://github.com/wordpress-mobile/WordPress-iOS/issues/19640. I'll add a note to that issue.

Reply:
I pushed a [commit](https://github.com/wordpress-mobile/WordPress-iOS/pull/19693/commits/67b6aede7db8a5ad59207ff4b50009ea9bfe5d99) removing the track events from `getWordPressInstallationState()` method and refactoring `MigratableStateTracker` struct. But the `MigratableStateTracker` is not used anywhere.

Based on the Github issue @twstokes shared, I understand that there will be an upcoming task to track those events, is that correct? If so, perhaps we can address calling those tracks in another PR. 

![CleanShot 2022-12-01 at 20 36 15@2x](https://user-images.githubusercontent.com/9609223/205143873-1a279da3-3c2f-4461-9947-ff95d0243556.png)

Reply:
Sounds good!

Reply:
> I'm assuming events tracking should be handled in a different PR because right now, we're not handling the case of "WP data ready to import" is "No" yet ( from the flowchart ).

Yep! That's the plan @salimbraksa. 👍